apps/mobile/src/features/token/address-list.tsx

```diff
@@ -0,0 +1,192 @@
+import { useRef, useState } from 'react';
+
+import { Balance } from '@/components/balance/balance';
+import { useCopyAddress } from '@/hooks/use-copy-address';
+import { Account } from '@/store/accounts/accounts';
+import { useAccountByIndex } from '@/store/accounts/accounts.read';
+import { useBitcoinPayerAddressFromAccountIndex } from '@/store/keychains/bitcoin/bitcoin-keychains.read';
+import { useStacksSignerAddressFromAccountIndex } from '@/store/keychains/stacks/stacks-keychains.read';
+import { t } from '@lingui/macro';
+
+import { Money } from '@leather.io/models';
+import { Box, Cell, SheetRef, Text } from '@leather.io/ui/native';
+import { truncateMiddle } from '@leather.io/utils';
+
+import { AccountAvatar } from '../account/components/account-avatar';
+import { getAssets } from '../receive/get-assets';
+import { SelectedAsset } from '../receive/screens/select-asset';
+import { TokenDetailsCard } from './components/token-details-card';
+import { useGetAccountTokenBalance } from './hooks/use-get-token-balance';
+import { ReceiveSheet, ReceiveSheetData } from './token-receive-sheet';
+
+export function AccountAddressList({
+  tokenId,
+  accountIndex,
+  fingerprint,
+}: {
+  tokenId: string;
+  accountIndex: number;
+  fingerprint: string;
+}) {
+  const account = useAccountByIndex(fingerprint, accountIndex);
+  if (!account) {
+    return null;
+  }
+
+  return (
+    <TokenDetailsCard
+      title={
+        <Box flexDirection="row" alignItems="center" gap="2">
+          {/* FIXME - this icon needs to be 16x16 but require refactors to make it work */}
+          <AccountAvatar icon={account.icon} />
+          <Text variant="label03">{account.name}</Text>
+        </Box>
+      }
+    >
+      <AddressList account={account} tokenId={tokenId} />
+    </TokenDetailsCard>
+  );
+}
+
+function AddressList({ account, tokenId }: { account: Account; tokenId: string }) {
+  const { nativeSegwitPayerAddress, taprootPayerAddress } = useBitcoinPayerAddressFromAccountIndex(
+    account.fingerprint,
+    account.accountIndex
+  );
+  const stxAddress = useStacksSignerAddressFromAccountIndex(
+    account.fingerprint,
+    account.accountIndex
+  );
+
+  const assets = getAssets({
+    nativeSegwitPayerAddress,
+    taprootPayerAddress,
+    stxAddress: stxAddress ?? '',
+  });
+
+  const tokenBalance = useGetAccountTokenBalance({ tokenId, account });
+  const availableBalance = tokenBalance?.availableBalance;
+  const quoteBalance = tokenBalance?.quoteBalance;
+
+  if (tokenId === 'BTC') {
```

Comment:
This balance section of this code will be re-written once we have balances split by BTC address type. 

For now it's using `useGetAccountTokenBalance` which will be removed and has both BTC balance listed as the same aggregate balance:
![Uploading Screenshot 2025-07-17 at 19.30.23.png…]()